src/scores/probability/fss_impl.py

```diff
@@ -0,0 +1,528 @@
+"""
+This module contains methods related to the FSS score
+
+For an explanation of the FSS, and implementation considerations,
+see: [Fast calculation of the Fractions Skill Score][fss_ref]
+
+[fss_ref]:
+https://www.researchgate.net/publication/269222763_Fast_calculation_of_the_Fractions_Skill_Score
+"""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import TYPE_CHECKING, Iterable, Optional, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+import xarray as xr
+
+from scores import utils
+from scores.typing import XarrayLike
+
+# Note: soft keyword `type` only support from >=3.10
+# "tuple" of 64 bit floats
+f8x3 = np.dtype("f8, f8, f8")
+# Note: `TypeAlias` on variables for python <=3.9
+if TYPE_CHECKING:  # pragma: no cover
+    FssDecomposed = Union[np.ArrayLike, np.DtypeLike]
+else:  # pragma: no cover
+    FssDecomposed = npt.NDArray[f8x3]
+
+
+class DimensionError(ValueError):
+    """
+    Wrapper for specific coordinate/dimension mismatch errors
+    """
+
+    # better to be explicit because python doesn't have braces for scoping its
+    # control flow
+    pass  # pylint: disable=unnecessary-pass
+
+
+class FssComputeMethod(Enum):
+    """
+    Choice of compute backend for FSS.
+    - Standard
+        - NUMPY (default)
+    - Optimized
+        - NUMBA_NATIVE (unavailable)
+        - NUMBA_PARALLEL (unavailable)
+    - Experimental
+        - RUST_NATIVE (unavailable)
+        - RUST_OCL (unavailable)
+    """
+
+    NUMPY = 1  # (default)
+    NUMBA_NATIVE = 2
+    NUMBA_PARALLEL = 3
+    RUST_NATIVE = 4
+    RUST_OCL = 5
+
+
+def fss_2d(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    threshold: np.float64,
```

Comment:
What is this threshold. Prefer a slightly more descriptive variable name.

Reply:
Its to bisect the fields into 0s and 1s. In the reference papers it's used in precipitation for example, so I guess `event_threshold` makes sense? 

If I'm understanding correctly - if we want to compute FSS for rainfall events > X mm, X would be the threshold.

Reply:
Yes, event_threshold sounds good. You could maybe even use the event operator from the binary contingency work to produce the event data. That would be a nice piece of consistency, and also allow the provision of more complex event definitions.

Reply:
I had a look at that, but it seems to support a lot more scope than a simple operation. Maybe I could break it up into a different issue, since I'll have to look at it more carefully.

Reply:
Well, it seems useful at least to allow simple operations like greater-than-or-equal, greater-than, less-than, less-than-or-equal. These can be accessed from the python 'operator' module. Let's review tomorrow whether to use the full event operator class or just allow simple operator specification.

Reply:
The other reasoning I think I had is that I didn't want to overload the `ufunc` calls with additional function decoration. I don't think python `inlines`  things very well unless its explicit. `numpy` knows to overload the `>` operator, but I'm not sure about custom operators, at the very least it'll have to unpack the wrapped operator.

Reply:
As discussed - will add `numpy.greater`, and it's siblings as optional threshold operator choices.